### ISSUE TITLE
Cocoon LED Strip (New Device Request??)

### ISSUE THREAD

**Issue by TheHomekitWayFinder** [AMoo-Miki/homebridge-tuya-lan#149]
I have just purchased a new [Cocoon Connect LED Strip](https://cocoonproducts.com.au/cocoonconnect/smart-led-light-strip/) from Aldi and have successfully acquired the I’d and key of the device and set the device up as a `SimpleLight` and `RGBTWLIGHT` but neither of them are working. I have got the schema and signature of the device to help with troubleshooting.

Schema:

` {
    name: 'LED Strip',
    id: '60574405cc50e37dcc5e',
    key: 'f698731bbfcb2a46',
    def: {
      schema: '[{"mode":"rw","code":"switch_led","name":"\\u5f00\\u5173","property":{"type":"bool"},"iconname":"icon-dp_power2","id":20,"type":"obj","desc":""},{"mode":"rw","code":"work_mode","name":"\\u6a21\\u5f0f","property":{"range":["white","colour","scene","music"],"type":"enum"},"iconname":"icon-dp_list","id":21,"type":"obj","desc":""},{"mode":"rw","code":"colour_data","name":"\\u5f69\\u5149","property":{"type":"string","maxlen":255},"iconname":"icon-dp_light","id":24,"type":"obj","desc":""},{"mode":"rw","code":"scene_data","name":"\\u573a\\u666f","property":{"type":"string","maxlen":255},"iconname":"icon-dp_box2","id":25,"type":"obj","desc":""},{"mode":"rw","code":"countdown","name":"\\u5012\\u8ba1\\u65f6\\u5269\\u4f59\\u65f6\\u95f4","property":{"unit":"","min":0,"max":86400,"scale":0,"step":1,"type":"value"},"iconname":"icon-timer1","id":26,"type":"obj","desc":""},{"mode":"wr","code":"control_data","name":"\\u8c03\\u8282","property":{"type":"string","maxlen":255},"iconname":"icon-setting","id":28,"type":"obj","desc":""}]',
      extras: '[{"id":20,"inputStyle":"","inputType":" "},{"id":21,"inputStyle":"","inputType":" "},{"id":26,"inputStyle":"","inputType":" "}]',
    },
  },
]`

Signature:

`[1/13/2020, 2:48:13 PM] [TuyaLan] Ready to handle LED Strip (RGBTWLight:3.3) with signature {"20":true,"21":"colour","24":"004303e80139","25":"000e0d00002e03e802cc00000000","26":0}`

I believe that “20” is power state, “21” is mode (not sure though), “24” is colour and brightness, not sure what “25” is or “26” and neither of these change.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Faisalthe01** [comment]
did you ever got it working?

**Comment by TheHomekitWayFinder** [comment]
No. I need to try on the newer version of the plug-in but haven’t yet got around to switching it over.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by TheHomekitWayFinder** [comment]
Try irayankhan/homebridge-tuya

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Faisalthe01** [comment]
hey mate, yeah I'm using that one but came across your post. Mine is working with the new plugin but Brightness is playing up. mine is missing a dp for brightness control just like you but you mentioned the 24 is for Color and Brightness both but it did not work for me

**Comment by TheHomekitWayFinder** [comment]
Yeah mines working but only for on off, colour and brightness doesn’t work

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by TheHomekitWayFinder** [comment]
I have stopped using this plugin as it is now broken because you cant get ids and keys for anything so if you can i would recommend using a tuya cloud plugin or the tuya integration in home assistant which is what i am doing and my LED strip works correctly
